stable/nginx-ingress/templates/controller-daemonset.yaml

```diff
@@ -12,6 +12,14 @@ metadata:
     release: {{ .Release.Name }}
   name: {{ template "nginx-ingress.controller.fullname" . }}
 spec:
+  selector:
+    matchLabels:
+      app: {{ template "nginx-ingress.name" . }}
+      component: "{{ .Values.controller.name }}"
+      release: {{ .Release.Name }}
+      {{- if .Values.controller.podLabels }}
+{{ toYaml .Values.controller.podLabels | indent 6}}
```

Comment:
Let's not add the podLabels to the selector since this field is immutable.

Reply:
According to [https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/#pod-selector](https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/#pod-selector):

> As of Kubernetes 1.8, you must specify a pod selector that matches the labels of the `.spec.template`. The pod selector will no longer be defaulted when left empty. Selector defaulting was not compatible with `kubectl apply`. Also, once a DaemonSet is created, its `.spec.selector` can not be mutated. Mutating the pod selector can lead to the unintentional orphaning of Pods, and it was found to be confusing to users.

Reply:
Make the selectors only be `app` and `release` which is pretty standard. As @taharah mentioned it is not a good idea to include the custom podLabels here.